src/screens/WhotoFollow/index.js

```diff
@@ -173,65 +173,70 @@ const WhotoFollow = () => {
-            setProfileId({
-              anonProfileId: anonymousUserId,
-              signedProfileId: userId,
-              token: res.token,
-              anonymousToken: res.anonymousToken
+      registerUser(data)
+        .then(async (res) => {
+          setFetchRegister(false);
+          if (res.code === 200) {
+            TokenStorage.set(res);
+            setToken(res.token);
+            try {
+              const userId = await JwtDecode(res.token).user_id;
+              const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
+              setProfileId({
+                anonProfileId: anonymousUserId,
+                signedProfileId: userId,
+                token: res.token,
+                anonymousToken: res.anonymousToken
+              });
+            } catch (e) {
+              crashlytics().recordError(new Error(e));
+            }
+            showMessage({
+              message: 'Welcome to Better Social',
+              type: 'success',
+              backgroundColor: colors.holytosca
+            });
+            setTimeout(() => {
+              create();
+              setImage(null, usersDispatch);
+              setInitialValue({id: res.token});
+            }, 2000);
+          } else {
+            crashlytics().recordError(new Error(res));
+            showMessage({
+              message: 'Cannot connect to server, please try again later',
+              type: 'danger',
+              backgroundColor: colors.red
             });
-          } catch (e) {
-            crashlytics().recordError(new Error(e));
           }
-          showMessage({
-            message: 'Welcome to Better Social',
-            type: 'success',
-            backgroundColor: colors.holytosca
-          });
-          setTimeout(() => {
-            create();
-            setImage(null, usersDispatch);
-            setInitialValue({id: res.token});
-          }, 2000);
-        } else {
-          crashlytics().recordError(new Error(res));
+        })
+        .catch((error) => {
+          crashlytics().recordError(new Error(error.response));
+          setFetchRegister(false);
           showMessage({
             message: 'Cannot connect to server, please try again later',
             type: 'danger',
             backgroundColor: colors.red
           });
-        }
-      })
-      .catch((error) => {
-        crashlytics().recordError(new Error(error.response));
-        setFetchRegister(false);
-        showMessage({
-          message: 'Cannot connect to server, please try again later',
-          type: 'danger',
-          backgroundColor: colors.red
         });
+    } catch (e) {
+      crashlytics().recordError(new Error(e));
+      setFetchRegister(false);
+      showMessage({
+        message: 'Cannot connect to server, please try again later',
+        type: 'danger',
+        backgroundColor: colors.red
       });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new changes have improved error handling by wrapping the entire registration process in a try-catch block. However, there's a potential issue with the error messages being too generic. The same error message is used for different types of errors (lines 216-219, 226-229, and 235-238). This could make it difficult to debug issues as they arise because the error message does not provide specific information about what went wrong.



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
No need this because we generalize the error message for the user